src/main/java/org/opensearch/security/dlic/rest/api/MigrateApiAction.java

```diff
@@ -102,6 +103,56 @@ private void migrateApiRequestHandlers(RequestHandler.RequestHandlersBuilder req
         requestHandlersBuilder.allMethodsNotImplemented().override(Method.POST, (channel, request, client) -> migrate(channel, client));
     }
 
+    protected boolean updateSecurityIndex(
+        final RestChannel channel,
+        final Client client,
+        ImmutableList.Builder<SecurityDynamicConfiguration<?>> builder
+    ) {
+        final List<SecurityDynamicConfiguration<?>> dynamicConfigurations = builder.build();
+        if (dynamicConfigurations.isEmpty()) {
+            LOGGER.info("Nothing to migrate");
+            return false;
+        }
+        final ImmutableList.Builder<String> cTypes = ImmutableList.builderWithExpectedSize(dynamicConfigurations.size());
+        final BulkRequestBuilder br = client.prepareBulk(securityApiDependencies.securityIndexName());
+        br.setRefreshPolicy(RefreshPolicy.IMMEDIATE);
+        try {
+            for (SecurityDynamicConfiguration<?> dynamicConfiguration : dynamicConfigurations) {
+                final String id = dynamicConfiguration.getCType().toLCString();
+                LOGGER.info("Updating " + id);
+                final BytesReference xContent = XContentHelper.toXContent(dynamicConfiguration, XContentType.JSON, false);
+                br.add(new IndexRequest().id(id).source(id, xContent));
+                cTypes.add(id);
+            }
+        } catch (final IOException e1) {
+            LOGGER.error("Unable to create bulk request " + e1, e1);
+            internalServerError(channel, "Unable to create bulk request.");
+            return false;
+        }
+
+        br.execute(new ConfigUpdatingActionListener<>(cTypes.build().toArray(new String[0]), client, new ActionListener<BulkResponse>() {
+
+            @Override
+            public void onResponse(BulkResponse response) {
+                if (response.hasFailures()) {
+                    LOGGER.error("Unable to upload migrated configuration because of " + response.buildFailureMessage());
+                    internalServerError(channel, "Unable to upload migrated configuration (bulk index failed).");
+                } else {
+                    LOGGER.debug("Migration completed");
+                    ok(channel, "Migration completed.");
+                }
+
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                LOGGER.error("Unable to upload migrated configuration because of " + e, e);
+                internalServerError(channel, "Unable to upload migrated configuration.");
+            }
```

Comment:
Quick question: Is this just a refactoring or are there functional changes in here?